scipy/special/_support_alternative_backends.py

```diff
@@ -40,15 +41,31 @@ def get_array_special_func(f_name, xp, n_array_args):
     def __f(*args, _f=_f, _xp=xp, **kwargs):
         array_args = args[:n_array_args]
         other_args = args[n_array_args:]
+
         if is_marray(_xp):
-            data_args = [np.asarray(arg.data) for arg in array_args]
-            out = _f(*data_args, *other_args, **kwargs)
+            data_args = [arg.data for arg in array_args]
+            f = globals()[f_name]  # Allow nested wrapping
+            out = f(*data_args, *other_args, **kwargs)
             mask = functools.reduce(operator.or_, (arg.mask for arg in array_args))
             return _xp.asarray(out, mask=mask)
+
+        elif is_dask(_xp):
+            f = globals()[f_name]  # Allow nested wrapping
+            # IMPORTANT: this works only because all ufuncs in this module
+            # are elementwise. It would be a grave mistake to apply this to gufuncs,
+            # as they would change their output depending on chunking!
+            return _xp.map_blocks(
+                # Hide other_args as well as kwargs such as dtype from map_blocks
+                lambda *array_args: f(*array_args, *other_args, **kwargs),
+                *array_args
+            )
+
         else:
+            assert array_args  # is_numpy(xp) is True
+            device = xp_device(array_args[0])
             array_args = [np.asarray(arg) for arg in array_args]
             out = _f(*array_args, *other_args, **kwargs)
-            return _xp.asarray(out)
+            return _xp.asarray(out, device=device)
```

Comment:
This device is parameter is untested, as only torch CPU and array-api-strict arrive here. So it only benefits an unknown non-CPU array with no guard against implicit device<->cpu transfers.